include/deal.II/base/template_constraints.h

```diff
@@ -25,10 +25,26 @@
 
 DEAL_II_NAMESPACE_OPEN
 
-namespace
+namespace internal
 {
-  // helper struct for is_base_of_all and all_same_as
-  template <bool... Types> struct BoolStorage;
+  namespace TemplateConstraints
```

Comment:
This would probably a good namespace name for everything in this file... Again, I don't quite see the need for a named namespace here.